src/EditorFeatures/Core.Wpf/InlineRename/UI/Adornment/RenameFlyoutViewModel.cs

```diff
@@ -60,13 +60,22 @@ internal class RenameFlyoutViewModel : INotifyPropertyChanged, IDisposable
             var smartRenameSession = smartRenameSessionFactory?.Value.CreateSmartRenameSession(_session.TriggerSpan);
             if (smartRenameSession is not null)
             {
-                SmartRenameViewModel = new SmartRenameViewModel(threadingContext, listenerProvider, smartRenameSession.Value);
+                SmartRenameViewModel = new SmartRenameViewModel(threadingContext, listenerProvider, smartRenameSession.Value, this);
                 SmartRenameViewModel.OnSelectedSuggestedNameChanged += OnSuggestedNameSelected;
+                SmartRenameViewModel.PropertyChanged += SmartRenameViewModel_PropertyChanged;
             }
 
             RegisterOleComponent();
         }
 
+        private void SmartRenameViewModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
```

Comment:
Could you check if `OnSuggestedNameSelected` and `SmartRenameViewModel_PropertyChanged` does a similar thing?
If so could they be merged? (Would prefer to use a special event instead of the general wpf property changing event to to this )